R/EXADBI-driver.R

```diff
@@ -0,0 +1,46 @@
+#' @include EXADBI-object.R
+NULL
+
+## EXADriver declaration and definition of driver instation functions.
+
+#' An interface driver object to the EXASOL Database.
+#'
+#' @seealso \code{\link{DBIDriver-class}}
+#' @family DBI classes
+#' @family EXADriver related objects
+#' @slot odbc_drv A string containing the path to the EXASOL ODBC driver file.
+#' @export
+#' @author EXASOL AG <opensource@exasol.com>
+setClass("EXADriver",
+         contains = c("DBIDriver", "EXAObject"),
+         slots = c(odbc_drv = "character")
+)
+
+#' Instantiates an EXADriver object.
+#' @family EXADriver related objects
+#' @param driver The path to an ODBC driver file. If missing, the driver installed on the system is used.
+#' If "SYSTEM": the EXASOL ODBC driver installed on the system is used immediately.
+#' Alternatively a path to an ODBC driver library can be provided.
+#' @param silent If TRUE, no message is print.
+#' @return An EXADriver object.
+#' @author EXASOL AG <opensource@exasol.com>
+#' @export
+exasol_driver <- function(driver = NULL, silent = FALSE) {
```

Comment:
This is a bit awkward.
In the original code this function was named exasol().
However R CMD check complains about a naming conflict with the package name! => So I renamed it to exasol_driver().

However the DBI::dbDriver() documentation it says:
```
dbDriver is a helper method used to create an new driver object given the name of a database or the corresponding R package. It works through convention: all DBI-extending packages should provide an exported object with the same name as the package. dbDriver just looks for this object in the right places: if you know what database you are connecting to, you should call the function directly.
```
According to our documentation we should not need a function with this name. But anyway it could cause any side effects.